engine/rules_engine.py

```diff
@@ -398,6 +399,56 @@ def validate_dataset_contents_against_library_metadata(
         # execute the rule
         return self.execute_rule(rule, dataset, dataset_path, datasets, domain)
 
+    def validate_variables_order_against_library_metadata(
+        self, rule: dict, dataset_path: str, datasets: List[dict], domain: str, **kwargs
+    ) -> List[Union[dict, str]]:
+        """
+        Validates variable order according to the library metadata.
+
+        Fetches the expected order from the library and
+        uses it as a comparator in the rule conditions.
+
+        Then, creates a df that maps variable name to its order like:
+        DOMAIN	SUBJECTID	AESEQ
+        1	    2	        3
+
+        Finally, executes the rule against the created df.
+        """
+        # get dataset class
+        dataset: pd.DataFrame = self.get_dataset_to_validate(
+            DatasetTypes.CONTENTS.value, dataset_path, datasets, domain
+        )
+        dataset_class: str = self.data_service.get_dataset_class(
+            dataset, dataset_path, datasets
+        )
+
+        # get metadata for this class and domain from the cache
+        domain_metadata: Optional[dict] = self._get_metadata_for_class_and_domain(
+            dataset_class, domain
+        )
+        if not domain_metadata:
+            raise ValueError(
+                f"Metadata for the following details is not found in CDISC Library: "
+                f"standard={self.standard}, version={self.standard_version}, class={dataset_class}, domain={domain}"
+            )
+
+        # fill rule conditions with expected variable ordinal
+        targets: List[str] = []
+        comparator: dict = {}
+        for variable in domain_metadata["datasetVariables"]:
+            targets.append(variable["name"])
+            comparator[variable["name"]] = variable["ordinal"]
+        self.rule_processor.create_list_of_conditions_for_each_target(rule, targets)
```

Comment:
I believe these methods should return something instead of modifying the rule, but that can be handled in a different work item if we want to go that route since you did not add that in this PR

Reply:
We already use these methods with the assumption that they change the rule by reference. We can of course change them to return a modified rule in a separate PR but let's first understand if this change benefits us